src/engraving/tests/eid_tests.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <gtest/gtest.h>
+
+#include "dom/engravingitem.h"
+
+#include "utils/scorerw.h"
+#include "utils/scorecomp.h"
+#include "utils/testutils.h"
+
+using namespace mu;
+using namespace mu::engraving;
+
+static const String DATA_DIR("all_elements_data/");
+
+class Engraving_EIDTests : public ::testing::Test
+{
+};
+
+static void itemShouldBeRegistered(void*, EngravingItem* item)
+{
+    if (item->registerId()) {
+        EngravingObject* registeredItem = item->masterScore()->eidRegister()->itemFromEID(item->eid());
+        EXPECT_TRUE(registeredItem);
+        EXPECT_EQ(registeredItem, item);
```

Comment:
Beware that GTest sometimes behaves weirdly when using these macros inside sub-methods rather than directly in the test method itself. Especially in sub-methods that return a value rather than void. Some ideas about how to handle this can be found here: https://stackoverflow.com/questions/53883790/check-for-errors-in-gtest-using-void-functions
However in this case it looks fine.